crates/randcast-mock-demo/src/main.rs

```diff
@@ -0,0 +1,239 @@
+
+    let msg = String::from("ujehwsndfgljkhrlkg");
+
+    println!("An user is requesting a randomness... msg seed: {}", msg);
+
+    let request_res = controller.request(msg.clone());
+
+    println!("request_res: {}", request_res);
+
+    println!("A signature task is emitting...");
+
+    let signature_task = controller.emit_signature_task();
+
+    let signature_index = signature_task.index;
+
+    // generates a partial sig with each share from the dkg
+    let partial_sigs = outputs
+        .iter()
+        .map(|output| G1Scheme::<BLS12_381>::partial_sign(&output.share, &msg.as_bytes()).unwrap())
+        .collect::<Vec<_>>();
+
+    // committer verify the partial threshold signatures first
+    partial_sigs.iter().for_each(|partial_sig| {
+        G1Scheme::<BLS12_381>::partial_verify(&public_poly, &msg.as_bytes(), &partial_sig).unwrap();
+    });
+
+    // then aggregates them
+    let sig = G1Scheme::<BLS12_381>::aggregate(t, &partial_sigs).unwrap();
+
+    // committer verify the threshold signature first
+    G1Scheme::<BLS12_381>::verify(&pubkey, &msg.as_bytes(), &sig).unwrap();
+
+    println!("Committers are committing result of the signature task...");
+
+    (0..n).for_each(|i| {
+        // the participant list to be rewarded by this signature task
+        let mut partial_signatures: HashMap<String, Vec<u8>> = HashMap::new();
+
+        partial_sigs
+            .iter()
+            .enumerate()
+            .for_each(|(i, partial_sig)| {
+                partial_signatures.insert(String::from("0x") + &i.to_string(), partial_sig.clone());
+            });
+
+        let res = controller.fulfill(
+            String::from("0x") + &i.to_string(),
+            signature_index,
+            sig.clone(),
+            partial_signatures,
+        );
+
+        println!("{}-res: {}", i, res);
+    });
+
+    let randomness_output = controller.get_last_output();
+
+    println!("randomness output: {}", randomness_output);
+
+    println!("finish.");
+
+    Ok(())
+}
+
+async fn run_dkg<C, S>(
+    board: &mut InMemoryBoard<C>,
+    phase0s: Vec<joint_feldman::DKG<C>>,
+    rng: &mut ThreadRng,
+) -> Vec<DKGOutput<C>>
+where
+    C: Curve,
+    // We need to bind the Curve's Point and Scalars to the Scheme
+    S: Scheme<Public = <C as Curve>::Point, Private = <C as Curve>::Scalar>,
+{
+    // Phase 1: Publishes shares
+    let mut phase1s = Vec::new();
+    for phase0 in phase0s {
+        phase1s.push(phase0.run(board, rng).await.unwrap());
+    }
+
+    // Get the shares from the board
+    let shares = board.shares.clone();
+
+    // Phase2
+    let mut phase2s = Vec::new();
+    for phase1 in phase1s {
+        phase2s.push(phase1.run(board, &shares).await.unwrap());
+    }
+
+    // Get the responses from the board
+    let responses = board.responses.clone();
+
+    let mut results = Vec::new();
+    for phase2 in phase2s {
+        results.push(phase2.run(board, &responses).await.unwrap());
```

Comment:
use `?`
refer: [Question Mark](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator).